code_challenges/advent_of_code/hydrothermal_venture.py

```diff
@@ -0,0 +1,138 @@
+def get_vents_input():
+    with open("code_challenges/advent_of_code/vents_input") as vents_input:
+        vents_coordinates = vents_input.read().split("\n")
+    # for vent_coordinate in vents_coordinates:
+    #     vent_coordinate.split("->")
+        
+    vents_coordinates = [vent_coordinate.split("->") for vent_coordinate in vents_coordinates]
+    vents_coordinates = [coordinate.strip() for vent_coordinate in vents_coordinates for coordinate in vent_coordinate]
+    vents_coordinates = [vent_coordinate.split(",") for vent_coordinate in vents_coordinates]
+    # vents_coordinates = [int(coordinate) for vent_coordinate in vents_coordinates for coordinate in vent_coordinate]
+    return vents_coordinates
```

Comment:
it is my understanding that i am getting a list of pairs. i just have them in a list not a tuple. does that matter?
[['0', '9'], ['5', '9'], ['8', '0'], ['0', '8'], ['9', '4'], ['3', '4'], ['2', '2'], ['2', '1'], ['7', '0'], ['7', '4'], ['6', '4'], ['2', '0'], ['0', '9'], ['2', '9'], ['3', '4'], ['1', '4'], ['0', '0'], ['8', '8'], ['5', '5'], ['8', '2']]

Reply:
This is a list of coordinates, not coordinate pairs.  The start and end point should be grouped together into a pair.

[(['0', '9'], ['5', '9']), (['8', '0'], ['0', '8']), (['9', '4'], ['3', '4']), .......]

Reply:
But also, if you don't need the properties of a list, you should use a tuple.

Reply:
oops you are right. i put them into pairs at the start of the function but could do that in the in its own function since i do it twice

Reply:
They are still not in pairs.  Your test for `get_coordinate_pairs` shows it.  It returns a flat list of coordinates, not a list of pair of coordinates. 

Also you should mock the call to `open` since you have so much logic in `get_vents_input`, then you can test it too.